pydantic_ai_slim/pydantic_ai/_output.py

```diff
@@ -98,15 +110,34 @@ def build(
         if output_type is str:
             return None
 
+        preferred_mode = None
         if isinstance(output_type, ToolOutput):
             # do we need to error on conflicts here? (DavidM): If this is internal maybe doesn't matter, if public, use overloads
             name = output_type.name
             description = output_type.description
             output_type_ = output_type.output_type
             strict = output_type.strict
+            preferred_mode = 'tool'
+        elif isinstance(output_type, StructuredOutput):
+            name = output_type.name  # TODO: Get this to the response_format model request arg
+            description = output_type.description  # TODO: Get this to the response_format model request arg
+            output_type_ = output_type.output_type
+            strict = output_type.strict  # TODO: Get this to the response_format model request arg
+            preferred_mode = 'structured'
         else:
             output_type_ = output_type
 
+        type_adapter = cast(TypeAdapter[T], TypeAdapter(output_type_))
+        json_schema = _utils.check_object_json_schema(type_adapter.json_schema(schema_generator=GenerateToolJsonSchema))
```

Comment:
If structured output requires an object schema, this is the place where you'll want to potentially incorporate or unify with the logic used by `pydantic_ai._output.OutputSchemaTool.__init__` to build the `response_data_typed_dict`. In particular, I wouldn't be surprised if you need to use `pydantic_ai._utils.is_model_like` in a similar way.

But maybe structured output doesn't have this restriction the way tools do?